include/tvm/runtime/threading_backend.h

```diff
@@ -0,0 +1,40 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file threading_backend.h
+ * \brief Utilities for manipulating thread pool threads.
+ */
+#ifndef TVM_RUNTIME_THREADING_BACKEND_H_
+#define TVM_RUNTIME_THREADING_BACKEND_H_
+
+#include <functional>
+#include <memory>
+#include <vector>
+
+namespace tvm {
+namespace runtime {
+namespace threading {
+
+class ThreadGroup {
+ public:
+  ThreadGroup();
```

Comment:
Why not blend Launch with the constructor or Create function. I do not see too much benefit of supporting multiple calls into Launch v.s. just Call ```  tgroup = ThreadGroup::Create(task_callbacks);```

We also need to document the behavior of Launch. Does it guarantee that the callbacks will be executed without relying on some previous callback to finish? For threadpool we need this because all worker needs to be alive. 

Normal thread executor do not have such requirement

Reply:
> do not see too much benefit of supporting multiple calls into Launch

I agree. The more I think about it, the more I think that _any_ launch should be MaxConcurrency.

> document the behavior of Launch

For sure. I was saving docs for after the code was throughly ~ripped into~ reviewed :)